### ISSUE TITLE
GraalWasm fails with data count section 12

### ISSUE THREAD

**Issue by willcohen** [oracle/graal#4729]
**Feature request**

Please include the following information:

**Is your feature request related to a problem? Please describe.**
Attempts to load a WASM file with a data count section fail with:
```
ExecutionError (PolyglotException) at org.graalvm.polyglot.Context/eval (Context.java:399):
invalid section ID: 12
```

**Describe the solution you'd like.**
Given that the spec declares that this section is [optional and useful mainly for a single-pass](https://webassembly.github.io/spec/core/binary/modules.html#binary-datacountsec), it seems to me that simply including it in [Section constants](https://github.com/oracle/graal/blob/master/wasm/src/org.graalvm.wasm/src/org/graalvm/wasm/constants/Section.java#L43-L55) and making it a simple additional path (that simply jumps to break, but provides a case beyond the default) in [BinaryParser](https://github.com/oracle/graal/blob/master/wasm/src/org.graalvm.wasm/src/org/graalvm/wasm/BinaryParser.java#L170-L174) to prevent crashing out would be sufficient.

**Describe who do you think will benefit the most.**
Anyone wishing to parse a file with a section 12.

**Describe alternatives you've considered.**
Transpiling the WASM to js and using graal.js, though with a performance hit.

**Additional context.**
I am working on trying to compile [PROJ](https://github.com/OSGeo/PROJ) via emscripten and would like to provide a WASM or JS fallback, so that JVM users can use it even if a native library isn't present for their platform. The built WASM works on the browser setting, but fails with (at least) this error on GraalWasm. I'm separately working on the alternative of transpiling but optimizing the JS beyond `-O0` is causing other issues in the interim, and will likely still be a less performant option even once that is resolved.

**Express whether you'd like to help contributing this feature**
If the fix is generally as simple as adding the new section to these two locations mentioned above, I'd be more than happy to assist! If it is a much larger lift, I'd still be happy to help but may need some guidance.

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for your suggestion, we will take a look into it and get back to you

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for your suggestion, we will take a look into it and get back to you

**Comment by j-blue-arz** [comment]
I ran into the same issue. It's already fixed in the code and available in the 23.0.0 developer build.
